tests/router_adv_with_m_flag/test_radv_mflag.py

```diff
@@ -0,0 +1,165 @@
+RADV_MAX_RA_INTERVAL_SECS = 4
+
+"""
+@summary: This fixture collects the data related to downlink VLAN port(s) and
+the connected PTF port(s) required to setup the RADV tests
+
+"""
+
+@pytest.fixture(scope="module", autouse=True)
+def radv_test_setup(request, duthosts, ptfhost, tbinfo):
+    duthost = duthosts[0]
+    logging.info("radv_test_setup() DUT {}".format(duthost.hostname))
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+
+    # RADVd is configured for each VLAN interface
+    vlan_dict = mg_facts['minigraph_vlans']
+    vlan_interfaces_list = []
+    for vlan_iface_name, vlan_info_dict in vlan_dict.items():
+        # Gather information about the downlink VLAN interface this relay agent is listening on
+        downlink_vlan_iface = {}
+        downlink_vlan_iface['name'] = vlan_iface_name
+
+        # Obtain the link-local IPv6 address of the DUT's downlink VLAN interface
+        downlink_vlan_iface['mac'] = duthost.get_dut_iface_mac(vlan_iface_name)
+        cmd = "ip -6 -o addr show dev {} scope link | sed -e's/^.*inet6 \([^ ]*\)\/.*$/\\1/;t;d'".format(vlan_iface_name)
+        res = duthost.shell(cmd)
+        ip6 = ipaddress.IPv6Address(unicode(res['stdout']))
+        pytest_assert(ip6.is_link_local,
+                "ip6 address:{} of {} is not a link-local address".format(str(ip6), downlink_vlan_iface['name']))
+        downlink_vlan_iface['ip6'] = str(ip6)
+
+        # Obtain link-local IPv6 address of the connected PTF port (Eg eth0)
+        # This PTF port maps to the first member of the TOR's VLAN
+        ptf_port = {}
+        ptf_port['port_idx'] = mg_facts['minigraph_ptf_indices'][vlan_info_dict['members'][0]]
+        ptf_port['name'] = "eth" + str(ptf_port['port_idx'])
+        cmd = "ip -6 -o addr show dev {} scope link | sed -e's/^.*inet6 \([^ ]*\)\/.*$/\\1/;t;d'".format(ptf_port['name'])
+        res = ptfhost.shell(cmd)
+        ip6 = ipaddress.IPv6Address(unicode(res['stdout']))
+        pytest_assert(ip6.is_link_local,
+                "ip6 address:{} of {} is not a link-local address".format(str(ip6), ptf_port['name']))
+        ptf_port['ip6'] = str(ip6)
+
+        vlan_intf_data = {}
+        vlan_intf_data['downlink_vlan_intf'] = downlink_vlan_iface
+        vlan_intf_data['ptf_port'] = ptf_port
+        vlan_interfaces_list.append(vlan_intf_data)
+
+    return vlan_interfaces_list
+
+"""
+@summary: Updates min/max RA interval in RADVd's config file
+
+"""
+
+def dut_update_ra_interval(duthost, ra, interval):
+    logging.info("Updating %s to %d in RADVd's config file:%s", ra, int(interval), RADV_CONF_FILE)
+    cmd = "sed -ie 's/\(.*\)\({}\) \([[:digit:]]\+\)/\\1\\2 {}/' {}".format(ra, interval, RADV_CONF_FILE)
+    duthost.shell("docker exec radv {}".format(cmd))
+
+"""
+@summary: A fixture that updates the RADVd's periodic RA update intervals and restores the
+intervals to old values after the test
+
+"""
+@pytest.fixture
+def dut_update_radv_periodic_ra_interval(duthost):
```

Comment:
Why are we updating radv interval here?

Reply:
For testing purpose I have updated radv interval, I will make the changes.